mutalyzer/website/views.py

```diff
@@ -673,28 +694,75 @@ def check_position(position, field):
-        return render_template('description-extractor.html')
+    There are two ways for the user to provide two sequences, corresponding to
+    the values for the `method` field, each requiring some additional fields to
+    be defined:
 
+    `method=raw_method`
+      The reference and sample sequences are pasted into the form fields.
+
+      - `reference_sequence`: The reference sequence.
+      - `variant_sequence`: The sample sequence.
+
+    `method=refseq_method`
+      The reference and sample sequences are given by RefSeq accession numbers.
+
+      - `reference_accession_number`: RefSeq accession number for the reference
+        sequence.
+      - `sample_accession_number`: RefSeq accession number for the sample
+        sequence.
+    """
     output = Output(__file__)
     output.addMessage(__file__, -1, 'INFO',
                       'Received Description Extract request from %s'
                       % request.remote_addr)
 
+    r = s = ''
+    reference_accession_number = request.args.get('reference_accession_number')
+    sample_accession_number = request.args.get('sample_accession_number')
+    reference_sequence = request.args.get('reference_sequence')
+    variant_sequence = request.args.get('variant_sequence')
+
+    method = request.args.get('method')
+    if method == 'refseq_method':
+        if not reference_accession_number:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Reference accession number input fields is empty.')
+        if not sample_accession_number:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Sample accession number input fields is empty.')
+
+        if reference_accession_number and sample_accession_number:
+            retriever = Retriever.GenBankRetriever(output)
+
+            reference_file = retriever.loadrecord(reference_accession_number)
+            sample_file = retriever.loadrecord(sample_accession_number)
+            if reference_file and sample_file:
+                r = unicode(reference_file.seq)
+                s = unicode(sample_file.seq)
+    else: # raw_method
+        if not reference_sequence:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Reference sequence number input fields is empty.')
+        if not variant_sequence:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Sample sequence number input fields is empty.')
+
+        if reference_sequence and variant_sequence:
+            r, s = reference_sequence, variant_sequence
+
     # Todo: Move this to the describe module.
-    if not util.is_dna(reference_sequence):
+    if not util.is_dna(r):
         output.addMessage(__file__, 3, 'ENODNA',
                           'Reference sequence is not DNA.')
-    if not util.is_dna(variant_sequence):
+    if not util.is_dna(s):
         output.addMessage(__file__, 3, 'ENODNA',
                           'Variant sequence is not DNA.')
 
```

Comment:
We'll need to set some limits on the input. For example, maximum sequence length and maximum difference in sequence length (e.g., shortest at least 20% of longest).

These limits should also be applied to the webservice interface.